odo/backends/tests/test_postgres.py

```diff
@@ -23,35 +23,46 @@
 null_data = [(1, None), (10, 20), (100, 200)]
 
 
+@pytest.fixture(scope='module')
+def tmpdir():
+    return os.environ.get('POSTGRES_TMP_DIR', None)
+
+
 @pytest.yield_fixture
-def csv():
+def csv(tmpdir):
     s = '\n'.join(','.join(map(str, row)) for row in data).encode('utf8')
-    with tmpfile('.csv') as fn:
+    with tmpfile('.csv', dir=tmpdir) as fn:
         with open(fn, 'wb') as f:
             f.write(s)
         yield CSV(fn)
 
 
 @pytest.yield_fixture
-def encoding_csv():
+def encoding_csv(tmpdir):
     path = os.path.join(os.path.dirname(__file__), 'encoding.csv')
-    with tmpfile('.csv') as fn:
+    with tmpfile('.csv', dir=tmpdir) as fn:
         with open(fn, 'wb') as f, open(path, 'r') as g:
             f.write(g.read().encode('latin1'))
         yield CSV(fn)
 
 
 @pytest.yield_fixture
-def complex_csv():
+def complex_csv(tmpdir):
     path = os.path.join(os.path.dirname(__file__), 'dummydata.csv')
-    with tmpfile('.csv') as fn:
+    with tmpfile('.csv', dir=tmpdir) as fn:
         shutil.copy(path, fn)
+        os.chmod(fn, 0777)
```

Comment:
This is a syntax error in py3, use `0o777` which is py2/3 comptible and imo more clear.